e2e/libs/volume/volume.py

```diff
@@ -101,6 +101,9 @@ def keep_writing_data(self, volume_name):
     def delete_replica(self, volume_name, node_name):
         return self.volume.delete_replica(volume_name, node_name)
 
+    def delete_unscheduled_replica(self, volume_name):
+        return self.volume.delete_replica(volume_name, "")
```

Comment:
Just to confirm, we are identifying the unscheduled replica which is without `node-name`, right? Is there any other flag which says replica is unscheduled?

Will there be any situation, where replica is taking more time to get scheduled and there are more than 1 replicas which are unscheduled?

Reply:
>  Is there any other flag which says replica is unscheduled?

Yes, it can probably also be identified by `address`, `currentImage`, `diskID`, `diskPath`, `instanceManagerName`, and `mode`:

```
replicas": [ 3 items

    {
        "address": "",
        "currentImage": "",
        "dataEngine": "v1",
        "dataPath": "replicas",
        "diskID": "",
        "diskPath": "",
        "failedAt": "",
        "hostId": "",
        "image": "longhornio/longhorn-engine:v1.7.0-rc3",
        "instanceManagerName": "",
        "mode": "",
        "name": "test-1-r-2adb049c",
        "running": false,
    },
    {
        "address": "10.42.3.9",
        "currentImage": "longhornio/longhorn-engine:v1.7.0-rc3",
        "dataEngine": "v1",
        "dataPath": "/var/lib/longhorn/replicas/test-1-f381acda",
        "diskID": "4d104962-2805-4d5f-b429-294e2318dd8e",
        "diskPath": "/var/lib/longhorn/",
        "failedAt": "",
        "hostId": "ip-10-0-1-180",
        "image": "longhornio/longhorn-engine:v1.7.0-rc3",
        "instanceManagerName": "instance-manager-bd9c950ca312e3542ca5efdc9a47244e",
        "mode": "RW",
        "name": "test-1-r-a4a745cd",
        "running": true,
    },
    {
        "address": "10.42.1.10",
        "currentImage": "longhornio/longhorn-engine:v1.7.0-rc3",
        "dataEngine": "v1",
        "dataPath": "/var/lib/longhorn/replicas/test-1-41d03df2",
        "diskID": "7691a424-da35-4c63-8553-0235c79bd16f",
        "diskPath": "/var/lib/longhorn/",
        "failedAt": "",
        "hostId": "ip-10-0-1-133",
        "image": "longhornio/longhorn-engine:v1.7.0-rc3",
        "instanceManagerName": "instance-manager-38078373032adc0641531cd38e9cd493",
        "mode": "RW",
        "name": "test-1-r-f73d6189",
        "running": true,
    },

],
```

> Will there be any situation, where replica is taking more time to get scheduled

When to run the `Delete unscheduled replica of ${workload_kind} ${workload_id} volume` step is decided by the person writing the test case. They should ensure the desired conditions are met before proceeding to the next test step.

> Will there be any situation, there are more than 1 replicas which are unscheduled?

It depends on the test case. If there are 2 nodes with scheduling disabled and a volume with 3 replicas is created, 2 replicas will remain unscheduled. At that point, the person writing the test case should decide whether to refine the implementation to meet their test requirements.

Reply:
What's a failed replica looks like? I guess, `running: false` would be same, what other parameter makes it different from `unscheduled `?

Reply:
I Removed this step since it makes no difference to the test case. Thank you!